include/aos/sm/networkmanager.hpp

```diff
@@ -538,13 +543,14 @@ class NetworkManager : public NetworkManagerItf {
     };
 
     using InstanceCache = StaticMap<StaticString<cInstanceIDLen>, NetworkData, cMaxNumInstances>;
-    using NetworkCache  = StaticMap<StaticString<cProviderIDLen>, InstanceCache, cMaxNumServiceProviders>;
+    using NetworkCache  = StaticMap<StaticString<cProviderIDLen>, SharedPtr<InstanceCache>, cMaxNumServiceProviders>;
```

Comment:
Looks like with fixed map emplace method we can avoid using shared ptr as well as mInstanceCacheAllocator.